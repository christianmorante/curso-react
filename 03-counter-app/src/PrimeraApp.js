import React from 'react'
import PropTypes from 'prop-types';

//import React, { Fragment } from 'react'


// Functional Components - FC
// const PrimeraApp = () => {
//     return ( 
//         <Fragment>
//             <h1>Hola Mundo</h1>
//             <p>Primeros pasos con React JS</p>
//         </Fragment>
//     );
// }

//const PrimeraApp = ({ saludo = 'Hola Mundo default' }) => {
const PrimeraApp = ({ saludo }) => {
    
    
    return ( 
        <>
            <h1>{ saludo }</h1>
            {/* <h1>{ JSON.stringify(saludo, null, 3) }</h1> */}
            <p>Primeros pasos con React JS</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

export default PrimeraApp;