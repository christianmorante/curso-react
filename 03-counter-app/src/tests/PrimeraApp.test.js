//import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';


describe('pruebas en PrimeraApp', () => {

    // test('should show message Hola soy Goku', () => {
    //     const saludo ='Hola soy Goku2';
    //     const { getByText } = render( <PrimeraApp saludo={ saludo } /> );
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // })

    test('should show PrimeraApp ok', () => {

        const saludo ='Hola soy Goku2';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> )
        expect( wrapper ).toMatchSnapshot();
    })
    
})
