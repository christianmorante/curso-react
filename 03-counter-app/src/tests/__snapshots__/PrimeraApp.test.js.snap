// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en PrimeraApp should show PrimeraApp ok 1`] = `
<Fragment>
  <h1>
    Hola soy Goku2
  </h1>
  <p>
    subtitulo por defecto
  </p>
</Fragment>
`;
